receiver_test.go

```diff
@@ -128,3 +132,123 @@ func TestReceiver_Nack_channel_ack_returns_error_it_should_return_the_same_error
 
 	assert.Same(t, expectedErr, err)
 }
+
+func TestReceiver_Receive(t *testing.T) {
+	dd := make(chan amqp.Delivery, 1)
+
+	now := time.Now()
+
+	dd <- amqp.Delivery{
+		Headers: amqp.Table{
+			"X-Custom-String": "test value",
+
+			"X-Custom-Int":   math.MinInt,
+			"X-Custom-Int8":  math.MinInt8,
+			"X-Custom-Int16": math.MinInt16,
+			"X-Custom-Int32": math.MinInt32,
+			"X-Custom-Int64": math.MinInt64,
+
+			"X-Custom-Float32": float32(math.MaxFloat32),
+			"X-Custom-Float64": math.MaxFloat64,
+
+			"X-Custom-String-Array": []string{"test value1", "test value2"},
+			"X-Custom-Int-Array":    []int{math.MinInt, math.MaxInt},
+			"X-Custom-Float-Array":  []float64{math.MaxFloat64},
+		},
+		ContentType:     "test content type",
+		ContentEncoding: "test content encoding",
+		CorrelationId:   "test correlation id",
+		ReplyTo:         "test reply to",
+		Expiration:      "test expiration",
+		MessageId:       "test message id",
+		Timestamp:       now,
+		Type:            "test message type",
+		UserId:          "test user id",
+		AppId:           "test app id",
+		ConsumerTag:     "test consumer tag",
+		DeliveryTag:     123,
+	}
+
+	close(dd)
+
+	ch := &mockChannel{}
+	ch.On("Consume", "test-queue", "test-consumer", false, false, false, false, amqp.Table(nil)).Return(dd, nil)
+	ch.On("Cancel", "test-consumer", false).Return(nil)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+
+	receiver := Receiver(ch, ConsumeOptions{
+		Queue:       "test-queue",
+		ConsumerTag: "test-consumer",
+	}, "test-alias")
+
+	rch, err := receiver.Receive(ctx)
+
+	assert.NoError(t, err)
```

Comment:
I suggested to also change it to `require.NoError`so the execution would be terminated, otherwise it will hang on the `nil` channel